x/ecocredit/server/location.go

```diff
@@ -0,0 +1,62 @@
+package server
+
+import (
+	"strings"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/pariz/gountries"
+)
+
+// validateLocation checks that the country and region conform to ISO 3166 and
+// the postal code is valid
+func validateLocation(location string) error {
+	countryCode, regionCode, postalCode, err := parseLocation(location)
+	if err != nil {
+		return err
+	}
+
+	iso3166query := gountries.New()
+
+	country, err := iso3166query.FindCountryByAlpha(countryCode)
```

Comment:
So I'd rather not check the list of countries and sub regions. This can change with little notice and we don't want to require a hard fork to update it. So for now I'd say we should just use a regex to validate 2 alphabetic characters for country and 1-3 alphanumeric characters for subregion. That's more future proof and doesn't introduce a possibly buggy third party dependency

Reply:
Yeah, makes sense